docs(cookbook): "Advanced Animations" cookbook recipe

DennisSmuda opened the pull request:
Closes #477

andretchen0 commented on docs/cookbook/advanced-animations.md:
```diff
@@ -0,0 +1,11 @@
+
+# Advanced Animations
+
+In this guide, we are goning to use GSAP to animate our scene instead of the `useRenderLoop` composable.
```
> goning

going

andretchen0 reviewed (changes requested):
Noticed a small spelling mistake.

andretchen0 reviewed (commented):
(Non-blocking)

Noticed a small spelling mistake, so I thought I'd flag it:

> goning

going

alvarosabu commented:
Thanks @DennisSmuda for creating this PR and the base for multiple languages 💚. Really looking forward for this one.

alvarosabu commented on docs/cookbook/advanced-animations.md:
```diff
@@ -8,4 +8,26 @@ difficulty: 0
 
-In this guide, we are goning to use GSAP to animate our scene instead of the `useRenderLoop` composable.
\ No newline at end of file
+In this guide, we are goning to use GSAP to animate our scene instead of the `useRenderLoop` composable.
```
```suggestion
In this guide, we are going to use GSAP to animate our scene instead of the `useRenderLoop` composable.
```

alvarosabu reviewed (changes requested):
Hey @DennisSmuda we would like to have this merge, could you please apply the suggestion and convert the PR to ready to review?

DennisSmuda commented:
Hey @alvarosabu sorry I had trouble getting to it 🗡️ 

English + German looks good to me now 👍

andretchen0 commented:
Oh, hey! Will review soon!

andretchen0 commented on docs/cookbook/advanced-animations.md:
````diff
@@ -0,0 +1,35 @@
+})
+```
+
+GSAP has been a very popular animation library and you can find a lot of help and resources online, including [these demos on codepen](https://codepen.io/GreenSock).
````
> GSAP has been a very

to

> GSAP is a very

andretchen0 commented on docs/public/recipes/advanced-animations.png:
Cool animation! Someone was looking over my shoulder and said the same thing. :)

I can't seem to get the animation to play in the docs. Here's what I see:

<img width="722" alt="Screenshot 2024-04-13 at 13 11 57" src="https://github.com/Tresjs/tres/assets/20469369/a5456fc5-ef68-4513-acbe-df7281a73ef0">

Maybe @alvarosabu can give you a hand getting that running on play.tresjs.org.

andretchen0 commented on docs/public/recipes/advanced-animations.png:
There are some elements that aren't doing anything in the demo, like the lights and shadow/alpha settings. We should remove those and their related imports.

andretchen0 commented on docs/public/recipes/advanced-animations.png:
(Non-blocking)

Some optional stuff:

The starting camera angle doesn't really show off what's happening:

<img width="581" alt="Screenshot 2024-04-13 at 13 31 59" src="https://github.com/Tresjs/tres/assets/20469369/586ea1e4-d64e-4721-8409-8f6fa2b64312">

Maybe pull back and to the side:

<img width="698" alt="Screenshot 2024-04-13 at 13 31 16" src="https://github.com/Tresjs/tres/assets/20469369/f247b8db-cd03-4f92-b576-054a623a9dd8">


* I'd remove the `Text3D` since it has to have a `Suspense` and that's covered elsewhere. And it's not necessary for the cool animation effect.
* I find `watch` to be more understandable at a glance than `watchEffect`. I like `watch` here because it shows beginners to TresJS + GSAP (like me!) what actually needs to be watched. (I didn't know.)
* And maybe more boxes just because GSAP makes that easy?

All that together:

```vue
<script setup lang="ts">
import { TresCanvas, useRenderLoop } from '@tresjs/core';
import { shallowRef, watch } from 'vue';
import { OrbitControls } from '@tresjs/cientos';
import gsap from 'gsap';

const boxesRef = shallowRef();
const zs = [];
for (let z = -4.5; z <= 4.5; z++) {
  zs.push(z);
}

watch(boxesRef, () => {
  //getting positions for all the boxes
  const positions = Array.from(boxesRef.value.children).map(
    (child) => child.position
  );
  //getting rotations for all the boxes
  const rotations = Array.from(boxesRef.value.children).map(
    (child) => child.rotation
  );

  const animProperties = {
    ease: 'power1.inOut',
    duration: 1,
    stagger: {
      each: 0.25,
      repeat: -1,
      yoyo: true,
    },
  };
  gsap.to(positions, {
    y: 2.5,
    ...animProperties,
  });
  gsap.to(rotations, {
    x: 2,
    ...animProperties,
  });
});
</script>

<template>
  <TresCanvas clear-color="#82DBC5">
    <TresPerspectiveCamera :position="[-15, 10, 15]" />
    <OrbitControls />
    <TresGroup ref="boxesRef">
      <TresMesh v-for="(z, i) of zs" :key="i" :position="[0, 0.5, z]">
        <TresBoxGeometry />
        <TresMeshNormalMaterial />
      </TresMesh>
    </TresGroup>
    <TresGridHelper :args="[10, 10, 0x444444, 'teal']" />
  </TresCanvas>
</template>
```

DennisSmuda replied:
Wow thanks so much for your thorough review @andretchen0 I'll get it done this week 👍

alvarosabu replied:
I agree with @andretchen0 in every point 😅

andretchen0 reviewed (changes requested):
Nice job! A few small changes. 

The long comment is non-blocking, so feel free to ignore or just take the useful-to-you bits.

DennisSmuda commented:
@andretchen0 I updated with your suggestions - thanks again ❤️ 

Also updated the screenshot for the overview:
<img width="659" alt="Screenshot 2024-04-23 at 23 17 08" src="https://github.com/Tresjs/tres/assets/6420600/b2aade41-d671-4726-a401-b63201a43ca9">

andretchen0 reviewed (approved):
Looks good to me! 👍 

@alvarosabu : ready to merge?

DennisSmuda commented:
> Looks good to me! 👍
> 
> @alvarosabu : ready to merge?

I only did german + english. I could remove the recipe from the other languages first, or generate with ChatGPT 😄

enpitsuLin commented:
Great work! I'm also a big fan of GSAP and dont known how it can work together with tres before this

@DennisSmuda I can review Chinese part if u want translate to other languages by LLVM, or I can raise a PR for Chinese after this be merged
